src/workerd/api/r2-test.js

```diff
@@ -674,6 +760,38 @@ export default {
           uploadedAfter: new Date('0'),
         },
       });
+      assert.equal(
+        await env.BUCKET.copy(
+          'onlyIfStrongEtag',
+          {
+            bucket: 'foo',
+            object: 'bar',
+          },
+          {
+            onlyIf: {
+              etagMatches: 'strongEtag',
+              etagDoesNotMatch: 'strongEtag',
+              uploadedBefore: new Date('0'),
+            },
+          }
+        ),
+        null
+      );
+      const multipart = await env.BUCKET.createMultipartUpload(
+        'onlyIfStrongEtagMultipart'
+      );
+      assert.rejects(
```

Comment:
The value returned from `rejects` needs to be awaited: https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message